### PR TITLE
Term API wrapper

### PR DESCRIPTION
**Pull request by MarcelDev**
ComputerCraft term api wrapper, allows for changing text colors, etc

[Documentation for term in ComputerCraft](https://tweaked.cc/module/term.html)

WARNING: british names are unimplemented along with [redirect](https://tweaked.cc/module/term.html#v:redirect), [current](https://tweaked.cc/module/term.html#v:redirect), [native](https://tweaked.cc/module/term.html#v:redirect)

### CODE REVIEW

File: colors.mw [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+: colors_pink # 64 ;
+: colors_gray # 128 ;
+: colors_lightGray # 256 ;
+: colors_cyan # 512 ;
+: colors_purple # 1024 ;
+: colors_blue # 2048 ;
+: colors_brown # 4096 ;
+: colors_green # 8192 ;
+: colors_red # 16384 ;
+: colors_black # 32768 ;
```
Comment (umnikos) [inline colors.mw]: maybe this would be better as 2^n instead of precomputed values?

File: colors.mw [hunk 2]
```diff
@@ -0,0 +1,18 @@
+" increasing fricken numbers " del
```
Comment (umnikos) [inline colors.mw]: unnecessary comment

File: examples/test_colors_term.mw
Comment (umnikos) [inline examples/test_colors_term.mw:1]: I think if we're going to be testing everything it's better for the tests to go in their own `tests/` folder

File: term.mw [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+    local x, y = term.getCursorPos()
+    push(x)
+    push(y)
+;;
+
+:: term_setCursorPos retry
+    local x = pop()
+    local y = pop()
+    term.setCursorPos(x, y)   
+;;
```
Comment (umnikos) [inline term.mw]: The order is inconsistent. With the current definition getting the position and then setting the position to the gotten position moves the cursor.

File: term.mw [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+
+:: term_getPaletteColor_rgb retry
+    local color = pop()
+    local r,g,b = term.getPaletteColor(color)
+
+    push(r)
+    push(g)
+    push(b)
+;;
```
Comment (umnikos) [inline term.mw]: Inconsistent argument order here as well

File: examples/test_custom_palette.mw [hunk 5]
```diff
@@ -0,0 +1,7 @@
+include term.mw
+include stdlib.mw
+include colors.mw
+
+colors_blue # 0x87cdff term_setPaletteColor_hex
```
Comment (umnikos) [inline examples/test_custom_palette.mw]: Wrong argument order given how `term_setPaletteColor_hex` is defined

File: term.mw (hunk 3 above)
Comment (umnikos) [reply term.mw]: Moonwalk currently doesn't have a rule for what order arguments and return values should go in for lua function wrappers, so how about we make them go in the same order they'd appear in lua? (right being top of the stack)

**Review by umnikos — CHANGES_REQUESTED** [review]

File: colors.mw (hunk 1 above)
Comment (MarcelDev) [reply colors.mw]: why?

File: colors.mw (hunk 2 above)
Comment (MarcelDev) [reply colors.mw]: yea lol

File: examples/test_colors_term.mw
Comment (MarcelDev) [reply examples/test_colors_term.mw:1]: done

File: term.mw (hunk 3 above)
Comment (MarcelDev) [reply term.mw]: reversed some stuff

File: examples/test_custom_palette.mw (hunk 5 above)
Comment (MarcelDev) [reply examples/test_custom_palette.mw]: fixed

File: term.mw (hunk 4 above)
Comment (MarcelDev) [reply term.mw]: also fixed

File: lib/term.mw [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+
+:: term_scroll retry
+    local y = pop()
+    term.scroll(y)
+;;
+
+:: term_getCursorPos retry
+    local x, y = term.getCursorPos()
+    push(y)
+    push(x)
```
Comment (umnikos) [inline lib/term.mw]: These are still backwards. You should push x before y so that y remains the top (aka. rightmost) value.

File: lib/term.mw [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+
+    term.blit(text, textColor, backgroundColor)
+;;
+
+:: term_setPaletteColor_hex retry
+    local index = pop()
+    local color = pop()
+
+    term.setPaletteColor(index, color)
+;;
```
Comment (umnikos) [inline lib/term.mw]: Also backwards, which makes the custom palette test fail, too

**Review by umnikos — CHANGES_REQUESTED** [review]

File: lib/term.mw [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+
+:: term_isColor retry
+    push(term.isColor())
+;;
+
+:: term_blit retry
+    local backgroundColor = pop()
+    local textColor = pop()
+    local textColor = pop()
```
Comment (umnikos) [inline lib/term.mw:73]: what

File: lib/term.mw [hunk 9]
```diff
@@ -0,0 +1,101 @@
...
+;;
+
+:: term_getPaletteColor_rgb retry
+    local color = pop()
+    local r,g,b = term.getPaletteColor(color)
+
+    push(b)
+    push(g)
+    push(r)
+;;
```
Comment (umnikos) [inline lib/term.mw:101]: incorrect orders

**Pull request merged**